src/shallowClone.ts

```diff
@@ -0,0 +1,9 @@
+export const shallowClone = <T extends object>(source: T): T => {
+  const base = Array.isArray(source) ? [] : {};
+  // TODO(asif): Remove this hack when TS/pull/13288 is merged in
+  const sourceObj = source as {} as object;
```

Comment:
what's the syntax equivalent to in JS?  I'm not familiar w/ it

Reply:
It's just const sourceObj = source;
I had to add the as {} as object to make it 'forget' its typing until that bug is resolved.

Reply:
ah cool thanks